code/DDSCodeTester.cpp

```diff
@@ -3285,6 +3285,93 @@ void dds_dataWriter_examples()
             //!
         }
     }
+
+    {
+        //DDS_DATAWRITER_SAMPLE_PREFILTER
+        struct CustomUserWriteData : public eprosima::fastdds::rtps::WriteParams::UserWriteData
+        {
+            CustomUserWriteData(
+                    const eprosima::fastdds::rtps::GuidPrefix_t& prefix)
+                : filtered_out_prefix(prefix)
+            {
+            }
+
+            eprosima::fastdds::rtps::GuidPrefix_t filtered_out_prefix;
+        };
+
+        struct CustomPreFilter : public eprosima::fastdds::dds::IContentFilter
+        {
+            // Custom evaluation
+            bool evaluate(
+                    const SerializedPayload& payload,
+                    const FilterSampleInfo& filter_sample_info,
```

Comment:
These are not used. Either comment them, or put `static_cast<void>()` calls in the body of the method